core/01_core.js

```diff
@@ -455,6 +455,65 @@
     op_is_weak_set,
   } = ensureFastOps();
 
+  function propWritable(value) {
+    return {
+      value,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    };
+  }
+
+  function propNonEnumerable(value) {
+    return {
+      value,
+      writable: true,
+      enumerable: false,
+      configurable: true,
+    };
+  }
+
+  function propReadOnly(value) {
+    return {
+      value,
+      enumerable: true,
+      writable: false,
+      configurable: true,
+    };
+  }
+
+  function propGetterOnly(getter) {
+    return {
+      get: getter,
+      set() {},
+      enumerable: true,
+      configurable: true,
+    };
+  }
+
+  function propNonEnumerableLazyLoaded(getter, loadFn) {
```

Comment:
i wonder if these should rather be a function that returns a function, where the outer function takes the loadFn, since you'd pass a loadFn to every call, which for cases ie webgpu, is quite a lot. though we could also just create a function on the use-site if necessary enough.

On the topic of load functions: its annoying that we have to create an identical load fn just with a different holding variable for every lazy loadable module, it would be nice if we had a utility function for that. maybe something along the lines of:
```js
function createLazyLoader(specifier) {
  let value;
  
  return function lazyLoad() {
    if (!value) {
      value = ops.lazy_load_esm(specifier);
    }
    return value;
  }
}